### PR TITLE
Miner was blocked when miner accepted an invalid block

### LINKED ISSUE #290: Other nodes are blocked when miner drug out an illegal node
**Brief of your idea**
The logic in doMiner() is that first of all, we need to do a loop to comprare utxo's lastedBlockID  with ledger's lastedBlockID，it will exit only if the two values are equal.
Other nodes are blocked when miner drug out an illegal node

**What could be better if XuperChain using your idea**
We plan to add a switch, if this switch is true, we'll be continue to keep old logic; not, we'll remove this logic. So the process are not blocked.

**Are there any side effects?**
Describe what might be worse.

**Additional information**
If possible, include a minimal, complete, and verifiable example to help us identify the issue.

### PR DESCRIPTION
**Pull request by MichelleZ**
## Description

What is the purpose of the change?
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #290

## Type of change

Please delete options that are not relevant.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Brief of your solution
We plan to add a switch, if this switch is true, we'll be continue to keep old logic; not, we'll remove this logic. So the process are not blocked.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

### CODE REVIEW

File: core/xchaincore.go [hunk 1]
```diff
@@ -503,16 +506,18 @@ func (xc *XChainCore) doMiner() {
...
-	for !bytes.Equal(ledgerLastID, utxovmLastID) {
-		xc.log.Warn("ledger last blockid is not equal utxovm last id")
-		err := xc.Utxovm.Walk(ledgerLastID)
-		if err != nil {
-			xc.log.Error("Walk error ", "ledger blockid", global.F(ledgerLastID),
-				"utxo blockid", global.F(utxovmLastID))
-			return
+	if xc.walkLoop {
+		for !bytes.Equal(ledgerLastID, utxovmLastID) {
+			xc.log.Warn("ledger last blockid is not equal utxovm last id")
```
Comment (fxsjy) [inline core/xchaincore.go]: (1) Here,  we should call "Walk" at least  once.  But, your code will cause Walk not be called if the flag is false
(2) the option flag 'walkLoop' can be changed to another word, such as "failover". Or, "failfast" in opposite.

File: ledger/ledger.go [hunk 2]
```diff
@@ -1027,3 +1027,29 @@ func (l *Ledger) MaxTxSizePerBlock() int {
...
+	}
+	newMeta.TrunkHeight = block.Height
+	metaBuf, pbErr := proto.Marshal(newMeta)
+	if pbErr != nil {
+		l.xlog.Warn("failed to marshal pb meta")
+		return pbErr
+	}
+	batchWrite.Put([]byte(pb.MetaTablePrefix), metaBuf)
+
+	l.meta = newMeta
```
Comment (fxsjy) [inline ledger/ledger.go]: No batchWrite.Write() ?

File: ledger/ledger.go [hunk 3]
```diff
@@ -1027,3 +1027,34 @@ func (l *Ledger) MaxTxSizePerBlock() int {
 func (l *Ledger) GetBaseDB() kvdb.Database {
 	return l.baseDB
 }
+
+func (l *Ledger) Truncate(ledgerLastID []byte, utxovmLastID []byte) error {
```
Comment (icexin) [inline ledger/ledger.go]: Is ledgerLastID redundant?

Comment (MichelleZ) [reply ledger/ledger.go]: The purpose of retaining this parameter is  to deleting  incorrect blocks in the future  for next upgrade

**Pull request merged**